graviti/utility/file.py

```diff
@@ -25,16 +26,50 @@ class File(RemoteFileMixin):
     __slots__ = ("url", "_checksum")
 
     def __init__(self, url: str, checksum: str) -> None:
-        super().__init__("")
-        self.url = READ_ONLY_URL(url)
+        self.url = url
         self._checksum = checksum
 
     def _repr_head(self) -> str:
         return f'{self.__class__.__name__}("{shorten(self._checksum)}")'
 
+    def _urlopen(self) -> UserResponse:
+        if not self.url:
+            raise ValueError(f"The file cannot open because {self._repr_head()} has no url")
+
+        session = get_session()
+        return UserResponse(
+            session.request("GET", self.get_url(), timeout=config.timeout, stream=True)
+        )
+
+    def get_url(self) -> str:
+        """Return the url of the data hosted by graviti.
+
+        Returns:
+            The url of the data.
+
+        Raises:
+            ValueError: When the url is missing.
+
+        """
+        if not self.url:
+            raise ValueError(f"The file URL cannot be got because {self._repr_head()} has no url")
+
+        return self.url
+
+    def open(self) -> Union[UserResponse, BufferedReader]:
+        """Return the binary file pointer of this file.
+
+        The remote file pointer will be obtained by ``urllib.request.urlopen()``.
```

Comment:
We do not use `urlopen()` now, we can remove this line.

Reply:
ok